origin-js/src/resources/marketplace.js

```diff
@@ -183,25 +201,23 @@ export default class Marketplace {
       loadOffers
     } = opts
 
-    const addOffersToListing = async (listing) => {
-      const offers = await this.getOffers(listingId, listing)
-      listing.offers = offers
-      return listing
-    }
-
     if (this.perfModeEnabled) {
       // In performance mode, fetch data from the discovery back-end to reduce latency.
       let listing = await this.discoveryService.getListing(listingId, blockInfo)
-      if (loadOffers)
-        listing = await addOffersToListing(listing)
+      if (loadOffers){
+        listing.id = listingId
```

Comment:
Why do we set listing.id only when loadoffers option is on ?
Also, why do we actually need to set this is since [initFromDiscovery](https://github.com/OriginProtocol/origin/blob/master/origin-js/src/models/listing.js#L75) should set it ?

Reply:
We need to set it because `_addOffersToListing` requires `listingId` in order to be able to query offers for a listing.

I see how code is currently confusing and not that much readable. Will add listingId as a function parameter to `_addOffersToListing` function to improve code readability.